de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/filesystem/IntelliJFileImplV2.java

```diff
@@ -174,7 +180,20 @@ public InputStream getContents() throws IOException {
             throw new FileNotFoundException(this + " does not exist or is " +
                 "derived");
 
-        return file.getInputStream();
+        Document document = Filesystem.runReadAction(
+            () -> FileDocumentManager.getInstance().getDocument(file));
+
+        if (document == null) {
```

Comment:
Is there any documentation when this might happen? It seems that there is no exception thrown, so if you did encounter `document` actually being `null`, I would like to see a comment about it.

Reply:
I don't think a comment is necessary in this case as the check is based on the [`FileDocumentManager.getDocument` javadoc](https://github.com/JetBrains/intellij-community/blob/36f1ada7ed36bfd72f854677fca58105fc6abd2a/platform/core-api/src/com/intellij/openapi/fileEditor/FileDocumentManager.java#L48):
```
@return the document, or null if the file represents a directory, or is binary without an associated decompiler, or is too large.
```

Reply:
Fair enough. That is easy to find in case somebody else might be wondering.